source/agora/registry/NameRegistryAPI.d

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+
+    Definitions of the name registry API
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.registry.NameRegistryAPI;
+
+import agora.common.Hash;
+import agora.common.Types;
+import agora.common.crypto.Key;
+
+import vibe.http.common;
+import vibe.web.rest;
+
+///
+struct RegistryPayloadData
+{
+    /// the public key that we want to register
+    public PublicKey public_key;
+
+    /// network addresses associated with the public key
+    public Address[] addresses;
+
+    /// monotonically increasing sequence number
+    public long seq;
+}
+
+///
+struct RegistryPayload
+{
+    ///
+    public RegistryPayloadData data;
+
+    /// signature over the `data` member
+    public Signature signature;
+
+    ///
+    public void sign_payload(const ref SecretKey secret_key)
+    {
+        signature = secret_key.sign(hashFull(data)[]);
+    }
+};
+
+/// API allowing to store network addresses under a public key
+static interface NameRegistryAPI
+{
+    @safe:
+
+    /***************************************************************************
+
+        Get network addresses corresponding to a public key
+
+        Params:
+            public_key = the public key that was used to register
+                         the network addresses
+
+        Returns:
+            Network addresses associated with the `public_key`
+
+        API:
+            GET /network_addresses
```

Comment:
Is that really a name we want ? I would suggest something like:
```suggestion
            GET /validators
```
OR:
```suggestion
            GET /nodes
```

`/addresses` would also be okay I guess, but the above feels more extensible (because it's tied to the content of the collection more than its type).